### PR TITLE
fix(typescript-estree): persisted parse and module none

### LINKED ISSUE #1502: Parsing errors when tsconfig.json's compilerOption's module is set to none
**What code were you trying to parse?**

```ts
// Parsing error: Cannot read property 'text' of undefined
const foo = 1;

// Parsing error: Cannot read property 'kind' of undefined
(() => { })();

// Parsing error: Cannot use 'in' operator to search for 'name' in undefined
let a: "a" | "b";
```

**What did you expect to happen?**
No parsing errors

**What actually happened?**
Got parsing errors

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.17.0` |
| `TypeScript`                | `3.7.5` |
| `ESLint`                    | `6.8.0` |
| `node`                      | `10.18.1` |
| `npm`                       | `6.13.6` |

### PR DESCRIPTION
**Pull request by armano2**
This is alternative (more extensive) change to how converter generates AST, this change simply removes needs of parent propagation for conversion logic. 

This change contains history from #1516, and its described in https://github.com/typescript-eslint/typescript-eslint/pull/1516#issuecomment-578369072

fixes #1502

### CODE REVIEW

File: packages/typescript-estree/src/create-program/createDefaultProgram.ts [hunk 1]
```diff
@@ -39,7 +39,10 @@ function createDefaultProgram(
     return undefined;
   }
 
-  const compilerHost = ts.createCompilerHost(commandLine.options, true);
+  const compilerHost = ts.createCompilerHost(
+    commandLine.options,
+    /* setParentNodes */ true,
```
Comment (bradzacher) [inline packages/typescript-estree/src/create-program/createDefaultProgram.ts]: do we need to set this now?

File: packages/typescript-estree/src/create-program/createIsolatedProgram.ts [hunk 2]
```diff
@@ -46,7 +46,7 @@ function createIsolatedProgram(code: string, extra: Extra): ASTAndProgram {
         filename,
         code,
         ts.ScriptTarget.Latest,
-        true,
+        /* setParentNodes */ true,
```
Comment (bradzacher) [inline packages/typescript-estree/src/create-program/createIsolatedProgram.ts]: ditto

File: packages/typescript-estree/src/create-program/createDefaultProgram.ts (hunk 1 above)
Comment (armano2) [reply packages/typescript-estree/src/create-program/createDefaultProgram.ts]: no, its not longer needed

**Comment by armano2** [comment]
make sure to validate all lines if i'm passing correct property as parent (i know its huge)

**Comment by bradzacher** [comment]
I eyeballed it all and it looked fine.
I'd hope that the tests would break if there was something you got wrong.
Seeing green so I think you got it all.

**Comment by bradzacher** [comment]
TBF as well, there's only 6 instances of `parent.` in `convert.ts`.
So there's only a few cases to check.
Might be worth triple checking that there are tests covering these cases

**Comment by armano2** [comment]
I did some testing, and it seems that we need parent nodes, than this change is not going to work, 

.parent is mostly used by tsutils and if i turn off parent nodes its starting to fail

**Comment by bradzacher** [comment]
😢 

That sucks. I didn't even think about the fact that tsutils uses parent.
Okay. reopen that and we can go with it instead.

**Comment by armano2** [comment]
than i'm closing this one as it will not work 😿

**Pull request closed without merging**